File: Examo_QA/examo-frontend/src/services/QuestionService.js
import axios from "axios";

const REST_API_BASE_URL = 'http://localhost:8080/api/questions';


export const listQuestions=() => axios.get(REST_API_BASE_URL);

export const createQuestion =(question) => axios.post(REST_API_BASE_URL, question);

export const getQuestion =(questionId) => axios.get(REST_API_BASE_URL +'/'+ questionId);

export const updateQuestion =(questionId, question) => axios.put(REST_API_BASE_URL +'/'+ questionId, question);

export const deleteQuestion =(questionId) => axios.delete(REST_API_BASE_URL+'/'+questionId);


export const getQuestionsByExamId = (examId) => axios.get(REST_API_BASE_URL + '/exam/' + examId);

File: examo-backend/examo-backend/src/main/java/com/examo/examo_backend/service/QuestionService.java
package com.examo.examo_backend.service;

import com.examo.examo_backend.dto.QuestionDto;
import com.examo.examo_backend.entity.Question;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;

import java.util.List;


public interface QuestionService {

    QuestionDto createQuestion(QuestionDto questionDto);

    QuestionDto getQuestionByQuestionId(Long questionId);

    List<QuestionDto> getAllQuestions();

    QuestionDto updateQuestion (Long questionId ,  QuestionDto updateQuestion);

    void deleteQuestion ( Long questionId );

    List<QuestionDto> getQuestionsByExamId(Long examId);



}
